week-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,93 @@
-const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
-];
 
 
-// Write your code here
\ No newline at end of file
+let element_image = document.getElementById('image');
+ let img_list = [
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+   "./assets/cute-cat-c.jpg",
+   "./assets/cute-cat-d.png"
+  ];
+element_image.setAttribute("src", img_list[0]);
+
+function add_carousel(){
+  let element_imageI = document.getElementById('image');
+  element_imageI.setAttribute("src", img_list.slice(-1)[0]);
+}
+
+function carousel(){
+  if(img_list) {
+    let element_imageJ = document.getElementsById('image');
+    element_imageJ.setAttribute("src", img_list[0]);
+  }else{
+    let element_imageK = document.getElementById('image');
+    element_imageK.setAttribute("src", img_list.slice(-1)[0])
+  }
+}
+
+function add_image() {
+  add_carousel();
+  carousel();
+}
+
+document.getElementById('previous').addEventListener('click', (event) => {
+    img_list.unshift(img_list.pop());
+  add_image();
+
+});
+
+document.getElementById('next').addEventListener('click', (event) => {
+img_list.push(img_list.shift());
```

Comment:
This is an interesting use of array methods to access the image and add it to the document. Consider simplifying by accessing the desired element in the  array with bracket notation